pkg/encoding/utils/pointsIO.go

```diff
@@ -243,6 +244,82 @@ func ReadG2Points(filepath string, n uint64, numWorker uint64) ([]bls.G2Point, e
 	start := uint64(0)
 	end := uint64(0)
 	size := n / numWorker
+	for i := uint64(0); i < numWorker; i++ {
+		start = i * size
+
+		if i == numWorker-1 {
+			end = n
+		} else {
+			end = (i + 1) * size
+		}
+		//todo: handle error?
+
+		go readG2Worker(buf, s2Outs, start, end, G2PointBytes, &wg)
+	}
+	wg.Wait()
+
+	// measure parsing time
+	t = time.Now()
+	elapsed = t.Sub(startTimer)
+	log.Println("    Parsing takes", elapsed)
+	return s2Outs, nil
+}
+
+func ReadG2PointSection(filepath string, from, to uint64, numWorker uint64) ([]bls.G2Point, error) {
+	g2f, err := os.Open(filepath)
+	if err != nil {
+		log.Println("ReadG2PointSection.ERR.0", err)
+		return nil, err
+	}
+
+	//todo: how to handle?
```

Comment:
@jianoaix , @ian-shim , is it necessary to handle this or can we just ignore it?